lib/Serialization/Deserialization.cpp

```diff
@@ -3376,37 +3376,58 @@ ModuleFile::getDeclCheckedImpl(DeclID DID) {
   case decls_block::PREFIX_OPERATOR_DECL: {
     IdentifierID nameID;
     DeclContextID contextID;
+    TypeID protoID;
 
-    decls_block::PrefixOperatorLayout::readRecord(scratch, nameID,
-                                                  contextID);
+    decls_block::PrefixOperatorLayout::readRecord(scratch, nameID, contextID,
+                                                  protoID);
     auto DC = getDeclContext(contextID);
-    declOrOffset = createDecl<PrefixOperatorDecl>(DC, SourceLoc(),
-                                                  getIdentifier(nameID),
-                                                  SourceLoc());
+    TypeLoc protoTyLoc = TypeLoc::withoutLoc(getType(protoID));
```

Comment:
Note that this has poor recovery (read: will crash) if the protocol can't be deserialized for whatever reason. That's probably fine for now, but really there should be a recovery mechanism.

Reply:
What's a better approach? We seem to be doing something very similar in other places.

Reply:
Look for the uses of `getTypeChecked` elsewhere in the file. Moving over to that for everything is a big task, but we probably want new fields to be thinking about recovery.